openapi.yaml

```diff
@@ -567,6 +692,140 @@ paths:
           $ref: "#/components/responses/InternalServerError"
 components:
   schemas:
+    IntegrateArgs:
+      type: object
+      properties:
+        provider:
+          $ref: '#/components/schemas/Intergrator'
+        requirer:
+          $ref: '#/components/schemas/Intergrator'
+      additionalProperties: false
+      required:
+        - provider
+        - requirer
+    Application:
+      type: object
+      description: Information about a given application.
+      properties:
+        charm-origin:
+          $ref: '#/components/schemas/CharmOrigin'
+        charm:
+          type: string
+        constraints:
+          $ref: '#/components/schemas/Constraints'
+        endpoint-bindings:
+          type: object
+          additionalProperties:
+            type: string
+        exposed:
+          type: boolean
+        exposed-endpoints:
+          type: object
+          additionalProperties:
+            $ref: '#/components/schemas/ExposedEndpoint'
+        principal:
+          type: boolean
+      additionalProperties: false
```

Comment:
There should be a list of resources returned here so that the deploy request can be followed by another to upload resources if required

POST /models/foo/applications/postgresql/resources/massive-file